### PR TITLE
Make HEARTBEAT handler ignore non-vehicle HEARTBEATs

### LINKED ISSUE #1187: HEARTBEAT handler failing to handle non-vehicle HEARTBEATs
Hey all: this is an issue with an associated PR (#1188) that I just put together.

I recently noticed a problem with dronekit and Herelink setups: basically, dronekit will attempt to parse every HEARTBEAT message that comes through, regardless of what type of system produced the message. This leads to issues, as Mavlink allows for non-vehicles to send HEARTBEAT messages (https://mavlink.io/en/messages/common.html#HEARTBEAT -- check out the `autopilot` field).

The symptom of this problem is a large number of dronekit APIExceptions, of the format (from herelinks):
`dronekit.APIException: mode (33554832, 0) not available on mavlink definition`
or
`dronekit.APIException: mode (0, 0) not available on mavlink definition`

I have confirmed that these messages are from the Herelink setup itself. This can be confirmed by checking the `mav.tlog` produced using mavproxy and cross-referencing the messages. (I can link a gist of the tlog if desired, but it doesn't really provide more info)

Anyways, this has been recently addressed in the `pymavlink` repository and a fix was merged into their master: https://github.com/ArduPilot/pymavlink/pull/769

I'm proposing a change (see PR) that uses mavutil's `probably_vehicle_heartbeat` function to ignore HEARTBEAT messages that don't come from vehicles.

This also relates to several other PRs/issues, such as #990,  but should be more all-encompassing and robust.

### PR DESCRIPTION
**Pull request by morzack**
Hi, this fixes #1187 . Check the issue for more details.

TL;DR (the two lines :) ): use mavutil's `probably_vehicle_heartbeat` function to avoid trying to parse non-vehicle HEARTBEATs.

Do note that this should **not** be merged in until pymavlink releases a new version. It relies on logic introduced in https://github.com/ArduPilot/pymavlink/pull/769 currently on their master.

### CODE REVIEW

**Comment by peterbarker** [comment]
Thanks for this!

Please ping me when you think we *should* merge this!

... we probably want something in the requirements.txt to require a version of pymavlink....

**Comment by morzack** [comment]
>Please ping me when you think we should merge this!

Ah yeah, I was wrong about there needing to be any waiting, the function already exists in the version of pymavlink used by dronekit. The new functionality won't be exposed until a new pymavlink release is made, however.

It can be merged at any time.

**Comment by morzack** [comment]
Hey! Just a quick follow-up since the most recent release of pymavlink implements the extra logic used by this commit. This is ready to be merged in whenever.

> ... we probably want something in the requirements.txt to require a version of pymavlink....

The minimum version of pymavlink specified in `requirements.txt` and `setup.py` has the `probably_vehicle_heartbeat()` function implemented, and pip will be default download the latest version of pymavlink when getting dronekit.

If you want, I can make a separate commit/PR to bump the minimum version of pymavlink to something newer, but that's not necessary for this PR to work.

@peterbarker if this looks good to you, feel free to merge when you get a chance. LMK if it needs changes.

**Comment by hamishwillee** [comment]
@peterbarker Ping ^^^^

**Comment by peterbarker** [comment]
Merged it, thanks!
